synfig-studio/src/gui/duckmatic.cpp

```diff
@@ -815,36 +815,27 @@ Duckmatic::set_guides_color(const synfig::Color &c)
 		signal_grid_changed().emit();
 	}
 }
-
-Duckmatic::GuideList::iterator
-Duckmatic::find_guide_x(synfig::Point pos, float radius)
+double
+Duckmatic::calculate_distance_from_guide(const Guide& guide, const Point& point)
 {
-	GuideList::iterator iter,best(guide_list_x_.end());
-	float dist(radius);
-	for(iter=guide_list_x_.begin();iter!=guide_list_x_.end();++iter)
-	{
-		float amount(std::fabs(*iter-pos[0]));
-		if(amount<dist)
-		{
-			dist=amount;
-			best=iter;
-		}
-	}
-	return best;
+	return std::fabs((cos(guide.angle.get())*(guide.point[1]-point[1])) - (sin(guide.angle.get())*(guide.point[0] - point[0])));
 }
 
+
 Duckmatic::GuideList::iterator
-Duckmatic::find_guide_y(synfig::Point pos, float radius)
+Duckmatic::find_guide(synfig::Point pos, float radius, Guide** second_best_guide_match)
```

Comment:
Can you add a description for this method?
I didn't read the entire discussion, so I didn't understand why the `second_best_guide_match` argument is needed.

Reply:
ok, I will add a comment describing it. But yes briefly it is needed in calculations to be able to snap to intersection of two guides.

p.s.
it wasn't needed in previous implementation because the behavior was simple if there is a horizontal guide snap the y coordinate of mouse and if there is a vertical snap the x coordinate of mouse which effectively ended up snapping to the intersection if present.